docker/docker_image_src.go

```diff
@@ -39,6 +39,19 @@ func newImageSource(ctx *types.SystemContext, ref dockerReference, requestedMani
 	if requestedManifestMIMETypes == nil {
 		requestedManifestMIMETypes = manifest.DefaultRequestedManifestMIMETypes
 	}
+	var (
+		mts   = supportedManifestMIMETypesMap()
+		reset = true
```

Comment:
(and as long you are changing this) this would benefit from either clearer variable names, or a comment explicitly describing what this loop intends to do. Preferably the former, perhaps naming them `supportedMIMEs` and `acceptableRequestedMIMEs` (with the opposite semantics to `reset`, starting as `false`).